Dockerfile

```diff
@@ -1,24 +1,40 @@
-FROM node:lts-alpine
-ENV NPM_CONFIG_LOGLEVEL warn
+FROM node:16.15.0-alpine
+
+ARG VERSION_INFO=noVersion
 
-# Install "git"
-RUN apk update \
- && apk add alpine-sdk
+ENV NPM_CONFIG_LOGLEVEL warn
+ENV PORT=1337
 
 # Set working dir as /app
 WORKDIR /app
 
 # Add sources to /app
-COPY . .
```

Comment:
This `COPY . .` step makes it so that the entire repo (minus a few gitignored files/folders) gets copied into the image. This forces unnecessary rebuilds of the image even on changes irrelevant to the functioning of the image, such as README changes.

Docker will cache each of these `COPY` instructions, so their ordering reflects how often they may change. Things that will change more often like `frontend` or `backend` are further down, and re-building the docker image will skip the other `COPY` instructions if the files they would copy are unchanged, saving time in the case of image rebuilds.